controllers/terminal/controllers/ingress.go

```diff
@@ -17,97 +17,87 @@ limitations under the License.
 package controllers
 
 import (
-	"bytes"
-	"text/template"
-
 	certv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	cmmeta "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
 	networkingv1 "k8s.io/api/networking/v1"
-	"k8s.io/apimachinery/pkg/util/yaml"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	terminalv1 "github.com/labring/sealos/controllers/terminal/api/v1"
 )
 
-const CertTmpl = `
-apiVersion: cert-manager.io/v1
-kind: Certificate
-metadata:
-  name: {{.ObjectMeta.Name}}
-  namespace: {{.ObjectMeta.Namespace}}
-spec:
-  secretName: {{.ObjectMeta.Name}}-terminal-sealos-io-cert
-  dnsNames:
-    - {{.ObjectMeta.Name}}.cloud.sealos.io
-  issuerRef:
-    name: cluster-issuer-terminal
-    kind: ClusterIssuer
-`
-
-const IngressTmpl = `
-apiVersion: networking.k8s.io/v1
-kind: Ingress
-metadata:
-  annotations:
-    kubernetes.io/ingress.class: nginx
-    nginx.ingress.kubernetes.io/rewrite-target: /
-    cert-manager.io/issuer: cluster-issuer-terminal
-  name: {{.ObjectMeta.Name}}
-  namespace: {{.ObjectMeta.Namespace}}
-  labels:
-    k8s-app: terminal
-spec:
-  rules:
-    - host: {{.ObjectMeta.Name}}.cloud.sealos.io
-      http:
-        paths:
-          - pathType: Prefix
-            path: /
-            backend:
-              service:
-                name: {{.ObjectMeta.Name}}
-                port:
-                  number: 8080
-  tls:
-    - hosts:
-        - {{.ObjectMeta.Name}}.cloud.sealos.io
-      secretName: {{.ObjectMeta.Name}}-terminal-sealos-io-cert
+const ClusterIssuerName = "cluster-issuer-terminal"
 
-`
-
-func parseTemplate(templateName string, terminal *terminalv1.Terminal) ([]byte, error) {
-	tmpl, err := template.New("test").Parse(templateName)
-	if err != nil {
-		return nil, err
+func createCert(terminal *terminalv1.Terminal) *certv1.Certificate {
+	objectMeta := metav1.ObjectMeta{
+		Name:      terminal.Name,
+		Namespace: terminal.Namespace,
 	}
-	buf := new(bytes.Buffer)
-	err = tmpl.Execute(buf, terminal)
-	if err != nil {
-		return nil, err
+	secretName := terminal.Name + "-terminal-sealos-io-cert"
+	dnsName := terminal.Name + ".cloud.sealos.io"
+	cert := &certv1.Certificate{
+		ObjectMeta: objectMeta,
+		Spec: certv1.CertificateSpec{
+			SecretName: secretName,
+			DNSNames:   []string{dnsName},
+			IssuerRef: cmmeta.ObjectReference{
+				Name: ClusterIssuerName,
+				Kind: "ClusterIssuer",
+			},
+		},
 	}
-	return buf.Bytes(), nil
+
+	return cert
 }
 
-func createCert(terminal *terminalv1.Terminal) (*certv1.Certificate, error) {
-	data, err := parseTemplate(CertTmpl, terminal)
-	if err != nil {
-		return nil, err
+func createIngress(terminal *terminalv1.Terminal) *networkingv1.Ingress {
+	objectMeta := metav1.ObjectMeta{
+		Name:      terminal.Name,
+		Namespace: terminal.Namespace,
+		Annotations: map[string]string{
+			"cert-manager.io/issuer":                     ClusterIssuerName,
+			"kubernetes.io/ingress.class":                "nginx",
```

Comment:
what if user using other ingress controller?

Reply:
service-auth/casdoor also use ingress-nginx, It would be better to support multiple ignress providers but as for now ingress-nginx will best match.